lib/jsdom/living/nodes/HTMLLabelElement-impl.js

```diff
@@ -1,12 +1,74 @@
 "use strict";
 
 const HTMLElementImpl = require("./HTMLElement-impl").implementation;
+const MouseEvent = require("../generated/MouseEvent");
 const closest = require("../helpers/traversal").closest;
+const domSymbolTree = require("../helpers/internal-constants").domSymbolTree;
+
+function isLabelable(node) {
+  /* labelable logic defined at: https://html.spec.whatwg.org/multipage/forms.html#category-label */
+  if (node.nodeType !== 1) { return false; }
+
+  switch (node.tagName) {
+  case 'BUTTON':
+  case 'KEYGEN':
+  case 'METER':
+  case 'OUTPUT':
+  case 'PROGRESS':
+  case 'SELECT':
+  case 'TEXTAREA':
+    return true;
+
+  case 'INPUT':
+    return node.type !== 'hidden' ? true : false;
+  }
+
+  return false;
+}
+
+function sendClickToAssociatedNode(node) {
+  node.dispatchEvent(
+    MouseEvent.createImpl(['click', {
```

Comment:
This should be `new MouseEvent` where `MouseEvent` is `require("../generated/MouseEvent").interface`, I think. Right, @Sebmaster?